apps/segment/src/modules/db/dynamo-config-repository.ts

```diff
@@ -0,0 +1,99 @@
+import { GetItemCommand, PutItemCommand } from "dynamodb-toolbox";
+import { err, ok, Result, ResultAsync } from "neverthrow";
+
+import { env } from "@/env";
+import { BaseError } from "@/errors";
+import { createLogger } from "@/logger";
+
+import { AppConfig } from "../configuration/app-config";
+import { DynamoConfigMapper } from "./dynamo-config-mapper";
+import { SegmentConfigEntityType, SegmentMainTable } from "./segment-main-table";
+import { ConfigRepository } from "./types";
+
+export class DynamoConfigRepository implements ConfigRepository {
+  private logger = createLogger("SegmentConfigRepository");
+  private mapper = new DynamoConfigMapper({
+    encryptionKey: env.SECRET_KEY,
+  });
+
+  static GetEntryError = BaseError.subclass("GetEntryError");
+  static SetEntryError = BaseError.subclass("SetEntryError");
+
+  constructor(
+    private deps: {
+      segmentConfigEntity: SegmentConfigEntityType;
```

Comment:
does it really have to be injected? In general DI is good, Im just wondering if this is not a false decoupling. Isn't it by design coupled? When can it be replaced with something else?

Reply:
I will check but I bet it won't be replaced